variables-cluster.tf

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+variable "create_cluster" {
+  default     = true
+  description = "Whether to create the OKE cluster and dependent resources."
+  type        = bool
+}
+
+variable "cluster_name" {
+  default     = "oke"
+  description = "The name of oke cluster."
+  type        = string
+}
+
+variable "control_plane_type" {
+  default     = "public"
+  description = "Whether to allow public or private access to the control plane endpoint"
+  type        = string
+
+  validation {
+    condition     = contains(["public", "private"], var.control_plane_type)
+    error_message = "Accepted values are public, or private."
+  }
+}
+
+variable "cni_type" {
+  default     = "flannel"
+  description = "The CNI for the cluster. Choose between flannel or npn"
+  type        = string
+  validation {
+    condition     = contains(["flannel", "npn"], var.cni_type)
+    error_message = "Accepted values are flannel or npn"
+  }
+}
+
+variable "kubernetes_version" {
+  default     = "v1.25.4"
+  description = "The version of kubernetes to use when provisioning OKE or to upgrade an existing OKE cluster to."
+  type        = string
+}
+
+variable "cluster_kms_key_id" {
+  default     = ""
+  description = "The id of the OCI KMS key to be used as the master encryption key for Kubernetes secrets encryption."
+  type        = string
+}
+
+variable "use_signed_images" {
+  description = "Whether to enforce the use of signed images. If set to true, at least 1 RSA key must be provided through image_signing_keys."
+  default     = false
+  type        = bool
+}
+
+variable "image_signing_keys" {
+  description = "A list of KMS key ids used by the worker nodes to verify signed images. The keys must use RSA algorithm."
+  type        = set(string)
```

Comment:
Is it possible to demand a set of keys if `use_signed_images` is set to `true`?

Reply:
Variable validations may only refer to themselves; I added a precondition in `cluster.tf`.